prog.go

```diff
@@ -88,6 +88,25 @@ func (ps *ProgramSpec) Copy() *ProgramSpec {
 	return &cpy
 }
 
+// RewriteMaps replaces all references to specific maps.
+//
+// Use this function to use pre-existing maps before loading the
+// program.
+func (ps *ProgramSpec) RewriteMaps(maps map[string]*Map) error {
+	for symbol, m := range maps {
+		fd := m.FD()
+		if err := ps.Instructions.RewriteMapPtr(symbol, fd); err != nil {
+			if asm.IsUnreferencedSymbol(err) {
```

Comment:
So in this case, we're explicitly rewriting the maps for a single program. Shouldn't this be an error then?